include/room_member.h

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Tunnler Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+//
+
+#pragma once
+
+#include <deque>
+#include <string>
+
+#include "tunnler.h"
+
+#include "RakNetStatistics.h"
+#include "RakPeerInterface.h"
+
+class RoomMember final {
+public:
+
+	struct MemberInformation {
+		std::string nickname;
+		std::string game_name;
+		MacAddress mac_address;
+	};
+
+	enum class State {
+
+		//Actual states
+		StateIdle, // Default state
+		StateError, // Some error [permissions to network device missing or something]
+		StateJoining, // This is also while the server didn't receive a setup packet / nickname from this player
+		StateJoined,
+    
+		// Reasons for connection loss
+		StateLostConnection,
+		StateKicked,
+    
+		// Reasons why connection was rejected
+		StateRoomFull,
+		StateRoomDestroyed,
+		StateWrongVersion,
+		StateWrongPassword,
+		StateBanned,
+		StatePasswordRequired,
+		StateNicknameAlreadyInUse,
+	};
+
+	struct ChatEntry {
+		std::string nickname;
+		std::string message;
+	};
+
+	RoomMember(MemberInformation member_information);
+	RoomMember(std::string nickname, std::string gameName, MacAddress mac_address);
+	~RoomMember();
+
+	State GetState();
+	MemberInformation GetMemberInformation();
+
+	// Retrieves the new chat lines since last calling this
+	std::deque<ChatEntry> PopChatEntries();
+
+	// Retrieves the new wifi packets filtered by PacketType and MacAddress since last calling this
+	std::deque<WifiPacket> PopWifiPackets(WifiPacket::PacketType type, const MacAddress& sender);
+
+	void SendChatMessage(std::string message);
+	void SendWifiPacket(const WifiPacket& packet);
+
+	//Returns some stats about the connection to the room
+	const std::string GetsStats();
+
+	//Returns the latest ping to the room
+	const int GetPing();
+
+	//Must be called periodically to receive new data
+	void Update();
+
+	void Join(std::string nickname, std::string server = "127.0.0.1", uint16_t serverPort = 1234, uint16_t clientPort = 0);
+	void Leave();
+	
+private:
+	State state;
+	MemberInformation member_information;
+
+	RakNet::RakPeerInterface* client;
+
+	static unsigned char GetPacketIdentifier(const RakNet::Packet* p);
```

Comment:
We should probably have an enum class of possible packet types, this is being discussed in the document right now.

Nvm, i didn't see the RoomMessages enum down there.